src/core/db/inc/persistent_data_interface.h

```diff
@@ -54,15 +58,24 @@ class database::DeviceDbInterface {
    * @brief Used to add an entry of the device data as part of
    *        registration process
    *
-   * @param mac Mac address of the IOT device (used as unique id)
-   * @param ip Ip address of the TCP server running on the IOT device
-   * @param port Port of the TCP server running on the IOT device
+   * @param id Unique ID of the IOT device (mac)
+   * @param link Link to the device for TCP communication
+   * @param type Type of IOT device
+   * @param title Title of the device
+   * @param description Textual description of the IOT device
    * @return true
    * @return false
    */
-  static bool addDevice(std::string mac, std::string ip, std::string port);
+  static bool add(std::string id, std::string link, std::string type,
+                  std::string title, std::string description);
+
+  static std::vector<radlet_dock::device::Device> getAll();
+
+  static radlet_dock::device::Device get(std::string);
+
```

Comment:
return shared_ptr pointers when data is non primitive if possible, explained below.